### DISCUSSION TITLE
Applied Operations Does not have ScaleIntensity

### DISCUSSION THREAD

**Discussion by a-parida12** [Project-MONAI/MONAI#5443] [Ideas]
When applying the scale intensity operation(which does linear scaling), it would be nice to have some meta information stored regarding the original intensity values(like. the original min and the original max). This would be helpful in getting back the original intensity ranges after some reconstruction problems(like thru VAE, Super Resolution etc.)

**Comment by KumoLiu** [comment]
Hi @a-parida12, I'm not sure if I understand correctly. Since you already have the original image why need to get back to it? And for the `Inverse` transform, it allows, for example, images to be cropped, rotated, padded, etc., during training and inference, and after being returned to their **original size** before saving to file for comparison in an external viewer.
Thanks!

**Reply by a-parida12** [reply]
@KumoLiu thanks for the quick feedback. To clarify here are some additional information.

I am interested in the `ScaleIntensity` functionality. In a general workflow you load an MRI, which could be of variety of ranges, lets say `[0, 1300]` using `ScaleIntensity` I rescale it to `[0, 1]`. I pass it through a Super Resolution which produces an output in range `[0, 1]` too. When I send it to medical practitioner he expects the range to  `[0, 1300]`. 

Currently, I re open the original  image  to get the range of the values. But it would be super nice to add the `ScaleIntensity` with an inverse too so helps prevent additional manual reading/tracking of original intensity ranges. Or at the least should have the original intensity values store in the meta information.

**Reply by KumoLiu** [reply]
@a-parida12, Oh I see. Maybe we can do something like sklearn. 
@wyli @Nic-Ma, What do you think?

https://github.com/scikit-learn/scikit-learn/blob/f3f51f9b611bf873bd5836748647221480071a87/sklearn/preprocessing/_data.py#L513-L534

**Reply by a-parida12** [reply]
Just to clarify it would be something like the `inverse` method in  https://docs.monai.io/en/stable/_modules/monai/transforms/spatial/array.html#Spacing ?

**Reply by AustinTapp** [reply]
Hi @a-parida12, @KumoLiu, @wyli, and @Nic-Ma. I was also wondering about an inverse for the ScaleIntensity transform. Could I ask what the status of this is, please?

Thank you!

**Reply by wyli** [reply]
There's another feature request here https://github.com/Project-MONAI/MONAI/issues/5647 perhaps optionally track these applied parameters is a good starting point.
